src/app/snap_to_grid.cpp

```diff
@@ -38,12 +38,21 @@ gfx::Point snap_to_grid(const gfx::Rect& grid,
       break;
 
     case PreferSnapTo::BoxOrigin:
+    case PreferSnapTo::FloorGrid:
       d = std::div(point.x-dx.rem, grid.w);
       newPoint.x = dx.rem + d.quot*grid.w;
 
       d = std::div(point.y-dy.rem, grid.h);
       newPoint.y = dy.rem + d.quot*grid.h;
       break;
+
+    case PreferSnapTo::CeilGrid:
```

Comment:
Not sure if this new `PreferSnapTo::CeilGrid` mode is required or when it's required. Do you have an image/test for this case? Any difference if we use this version in `DocExporter::captureSamples()`?
```c++
            gfx::Point posBottomRight =
              snap_to_grid(docPref.grid.bounds(),
                           frameBounds.point2()
                           + gfx::Point(docPref.grid.bounds().size()),
                           PreferSnapTo::FloorGrid);
```

Reply:
IMHO `PreferSnapTo::CeilGrid` is needed to guarantee the situation of a layer going up to the edge of a grid. For example, look at this sprite of 2 layers:

Layer 1:
![1_screenshot from 2019-02-06 00-09-51](https://user-images.githubusercontent.com/6051497/52310586-33168500-29a4-11e9-99fc-13301aff5e0c.png)

Layer 2:
![2_screenshot from 2019-02-06 00-10-08](https://user-images.githubusercontent.com/6051497/52310604-432e6480-29a4-11e9-96f0-9ba2dde606b5.png)

The expected output of "Trim by grid" should be (I think):
![3_screenshot from 2019-02-06 00-11-33](https://user-images.githubusercontent.com/6051497/52310627-52adad80-29a4-11e9-90d3-1771fa6b2d8f.png)

However, with the modification you have proposed it produces:
![4_screenshot from 2019-02-06 00-10-40](https://user-images.githubusercontent.com/6051497/52310640-60633300-29a4-11e9-8672-efa3367614b2.png)

This is because the layer 1 goes up to the edge of a grid, but there is no need of adding an extra one. Adding `+ gfx::Point(docPref.grid.bounds().size()),` in the `snap_to_grid` would add the extra grid gap that it is not needed. This is managed in `PreferSnapTo::CeilGrid:` asking if there is a remainder at the division `std::div(point.x-dx.rem, grid.w);` (it is the same in the y axis).

Reply:
Nice example 👍 I'll try to merge the PR today. LFTM